### DISCUSSION TITLE
Custom headers for airflow webserver

### DISCUSSION THREAD

**Discussion by sustarun** [apache/airflow#36981] [General]
### Description

I needed a way to add custom headers to the airflow webserver. I was able to do it but it required changes in the python code of airflow package which obviously is not the right practice. So wanted to get the functionality available at config level.

If this feature already exists or if there are any other solutions, please let me know.

### Use case/motivation

This feature arose out of a need to have security headers (like x-content-type-options, strict-transport-security) on airflow webserver that I had hosted on linux.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by dirrao** [comment]
@sustarun You are welcome to submit the MR.

**Comment by potiuk** [comment]
As discussed in discussion - this is something that should be done via proxy put in front of Airflow. We are note going to have all the possible bells-and-whistles that is usually provided by a dedicated proxy doing TLS termination with all the possible options such as limiting algorithms used and other specialized security behaviors, all kinds of security featyres such as rate limiting, setting up the security headers etc. Every configurability in this area we add to Airflow turns us closer to re-implementing what others (nginx, apache webserver and a dosen of dedicated proxy solutions) have already implemented. 

Converting it into a discussion, but we plan anyway to make it crystal clear in our documentation that in case you need such advanced features, as a Deployment Manager you are supposed to put properly configured proxy in front of airflow and use the features implemented there (basically standing on the shoulders of giants).
